File: metrics-extension/symfinder/src/main/java/fr/unice/i3s/sparks/deathstar3/symfinder/engine/configuration/Configuration.java
/*
 * This file is part of symfinder.
 *
 * symfinder is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * symfinder is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with symfinder. If not, see <http://www.gnu.org/licenses/>.
 *
 * Copyright 2018-2021 Johann Mortara <johann.mortara@univ-cotedazur.fr>
 * Copyright 2018-2021 Xhevahire Tërnava <t.xheva@gmail.com>
 * Copyright 2018-2021 Philippe Collet <philippe.collet@univ-cotedazur.fr>
 */

package fr.unice.i3s.sparks.deathstar3.engine.configuration;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;

import java.io.FileInputStream;
import java.io.IOException;

public class Configuration {

    private ParametersObject properties;

    public Configuration() {
        this("symfinder.yaml");
    }

    public Configuration(ParametersObject properties) {
        this.properties = properties;
    }

    private Configuration(String propertiesFile) {
        try (FileInputStream fis = new FileInputStream(propertiesFile)) {
            ObjectMapper objectMapperYaml = new ObjectMapper(new YAMLFactory());
            properties = objectMapperYaml.readValue(fis, ParametersObject.class);
        } catch (IOException e) {
            e.printStackTrace(); // TODO: 11/28/18 create exception
        }
    }

    public String getNeo4JBoltAddress() {
        return properties.neo4j().boltAddress();
    }

    public String getNeo4JUser() {
        return properties.neo4j().user();
    }

    public String getNeo4JPassword() {
        return properties.neo4j().password();
    }

    public int getSingularityThreshold() {
        return properties.hotspots().nbVariantsThreshold();
    }

    public int getAggregationThreshold() {
        return properties.hotspots().nbAggregationsThreshold();
    }

}

File: symfinder-ts/app/lib/configuration/Configuration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Configuration = void 0;
/*
 * This file is part of symfinder.
 *
 * symfinder is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * symfinder is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with symfinder. If not, see <http://www.gnu.org/licenses/>.
 *
 * Copyright 2021-2022 Bruel Martin <martin.bruel999@gmail.com>
 */
var js_yaml_1 = require("js-yaml");
var fs_1 = require("fs");
var Configuration = /** @class */ (function () {
    function Configuration(runner) {
        this.runner = runner;
        if (this.runner === "docker") {
            this.properties = (0, js_yaml_1.load)((0, fs_1.readFileSync)('./neo4j_docker.yaml', 'utf8'));
        }
        else {
            this.properties = (0, js_yaml_1.load)((0, fs_1.readFileSync)('./neo4j.yaml', 'utf8'));
        }
    }
    Configuration.prototype.getNeo4JBoltAdress = function () {
        return this.properties.neo4j.boltAddress;
    };
    Configuration.prototype.getNeo4JUser = function () {
        return this.properties.neo4j.user;
    };
    Configuration.prototype.getNeo4JPassword = function () {
        return this.properties.neo4j.password;
    };
    return Configuration;
}());
exports.Configuration = Configuration;
